WookieeMatt/Essence20
# Adding trait selector dialog; migration

Closes https://github.com/WookieeMatt/Essence20/issues/319

In this PR:
- Creating a new dialog for trait selection to replace checkboxes (borrowed from dnd5e)
- For now, trait selection only applies to Armor. There's an edit button on its item sheet to open the new dialog.
- Display the selected traits on the item sheet as chips
- Armor traits field is now an array of strings instead of a boolean map
- Migration to convert existing traits object (or in older cases, a single string) to an array of strings
- Dialog text size is small for some reason, open to ideas

Testing:
- 🚨 PLEASE TEST ON AN E20 WORLD YOU DON'T CARE ABOUT BREAKING 🚨
- In `beta`, create some armor with different traits selected
- Pull down this branch, `trait-selector`
- Be a pal and open devtools by pressing F12 before launching the test world. If something goes wrong, the logs could be helpful.
- Launch the test world
- You should see messages about the migration starting and eventually completing
- Armor traits should display as chips on the armor sheet
- Edit button should open the trait selector dialog
- Bonus: You can checkout a commit OLDER than `7465fdd` to create an armor with a single trait, and ensure that migration works too
